src/DnsResolver/custom/csharp/RestErrorHandler.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Management.Automation;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.PowerShell.Cmdlets.DnsResolver.Models.Api20220701;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DnsResolver.Cmdlets
+{
+    internal static class CmdletRestExtension
+    {
+        public static void WriteError(this Cmdlet cmdlet, HttpResponseMessage responseMessage, Task<ICloudError> errorResponseTask, ref Task<bool> returnNow)
+        {
+            var errorString = responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
We should avoid to print the whole response. According to your ask in the teams channel, can we just print the details section in the response.

Reply:
@dolauli : we need to print the **message + details**.

For the customer request validation, the details section includes all the missing things in the request payload.
For the business rule validation, the message includes as to why the validation failed.

Reply:
@arpit-gagneja, if there is no details section, you may re-use the logic in ondefault function. Besides we assume Azure services should follow the similar schema to return errors. And actually we have implemented the logic to desterilizing details from the error response, please see .\generated\api\Models\xxx\ErrorResponse.cs for details. So you may access the details by using something like `(await response)?.Detail;`

Reply:
thanks @dolauli for the suggestion. one more question here: I have made below changes and want to know if there is a utility function to convert Detail to string.

![image](https://user-images.githubusercontent.com/19483623/178381457-3987c46f-503f-4f73-ba66-eb8933ec1a1c.png)

Reply:
@dolauli : I have addressed the comment as suggested and tested the changes locally. The changes look fine.
Please review again at your earliest convenience.